### PR TITLE
Recurring Payments

### PR DESCRIPTION
**Pull request by gcatanese**
Implementation of the Recurring Payments use case.

The Checkout sample has been moved to `checkout-example` folder.
The Subscription sample has been implemented in the `subscription-example` folder.

### CODE REVIEW

File: subscription-example/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# Adyen [Tokenization](https://docs.adyen.com/online-payments-tokenization) Integration Demo
+
+This repository includes a tokenization example for subscriptions. Within this demo app, you'll find a simplified version of a website that offers a music subscription service. 
+The shopper can purchase a subscription and administrators can manage the saved (tokenized) payment methods on a separate admin panel. 
+The panel allows admins to make payments on behalf of the shopper using this token. We refer to this token as `recurringDetailReference` in this application. 
+
+## Workflow 
+
+The sample app implements the following workflow:
```
Comment (Kwok-he-Chu) [inline subscription-example/README.md:9]: Let's update this in the .NET one too

File: subscription-example/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+> You can find more information about webhooks in [this detailed blog post](https://www.adyen.com/blog/Integrating-webhooks-notifications-with-Adyen-Checkout).
+
+
+## Run integration on localhost using a proxy
+You will need Node.js 12+ to run this application locally.
+
+1. Clone this repository.
+
+```
+git clone https://github.com/adyen-examples/adyen-dotnet-online-payments.git
```
Comment (Kwok-he-Chu) [inline subscription-example/README.md:70]: wrong link here, should be node

File: subscription-example/index.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    helpers: require("./util/helpers"),
+  })
+);
+
+app.set("view engine", "handlebars");
+
+/* ################# API ENDPOINTS ###################### */
+
+// Invoke tokenization endpoint
+app.post("/tokenization/sessions", async (req, res) => {
```
Comment (Kwok-he-Chu) [inline subscription-example/index.js:52]: Add `/api/` prefix for consistency

File: subscription-example/package-lock.json
Comment (Kwok-he-Chu) [inline subscription-example/package-lock.json:1]: We can remove the lock.json

File: subscription-example/index.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  // Make Payment page
+  res.render("admin/makePayment", {
+    type: result,
+    recurringDetailReference: req.params.recurringDetailReference
+  })
+
+});
+
+// Invoke to disable a token
+app.get("/admin/disable/:recurringDetailReference", async (req, res) => {
```
Comment (Kwok-he-Chu) [inline subscription-example/index.js:165]: Question @gcatanese, do we want to make all calls have the `/api/`-prefix e.g. `/api/admin/disable`

File: subscription-example/public/adyenSubscriptionImplementation.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+// Used to finalize a checkout call in case of redirect
+const urlParams = new URLSearchParams(window.location.search);
+const sessionId = urlParams.get('sessionId'); // Unique identifier for the payment session
+const redirectResult = urlParams.get('redirectResult');
+
+async function startTokenization() {
+    const type = document.getElementById("type").innerHTML;
+    try {
+        const response = await callServer("/tokenization/sessions");
```
Comment (Kwok-he-Chu) [inline subscription-example/public/adyenSubscriptionImplementation.js:11]: Should we prefix here as well? `/api/tokenization/sessions/` (see comment above)

File: subscription-example/util/helpers.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
```
Comment (Kwok-he-Chu) [inline subscription-example/util/helpers.js:1]: Is this used? And if so, could you briefly add a comment what it does? :)

**Review by Kwok-he-Chu — COMMENTED** [review]
Nice work!

File: subscription-example/public/adyenSubscriptionImplementation.js (hunk 5 above)
Comment (gcatanese) [reply subscription-example/public/adyenSubscriptionImplementation.js:11]: Yes, to match the updated URL in `index.js`

File: subscription-example/index.js (hunk 3 above)
Comment (gcatanese) [reply subscription-example/index.js:52]: Done

File: subscription-example/README.md (hunk 2 above)
Comment (gcatanese) [reply subscription-example/README.md:70]: Corrected

File: subscription-example/index.js (hunk 4 above)
Comment (gcatanese) [reply subscription-example/index.js:165]: `/api` is used for API calls returning JSON, in this case the user is redirected to a different view

Comment (Kwok-he-Chu) [reply subscription-example/index.js:165]: Makes sense

**Review by Kwok-he-Chu — APPROVED** [review]
🤝

**Pull request merged**